sdk/internal/recording/example_test.go

```diff
@@ -0,0 +1,97 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package recording_test
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/recording"
+)
+
+func ExampleStart() {
```

Comment:
Could we also introduce an end-to-end example that shows how to use this framework?